packages/tailwindcss/src/canonicalize-candidates.ts

```diff
@@ -1110,6 +1112,516 @@ function* tryValueReplacements(
 
 // ----
 
+function isSingleSelector(ast: SelectorParser.SelectorAstNode[]): boolean {
+  return !ast.some((node) => node.kind === 'separator' && node.value.trim() === ',')
+}
+
+function isAttributeSelector(node: SelectorParser.SelectorAstNode): boolean {
+  let value = node.value.trim()
+  return node.kind === 'selector' && value[0] === '[' && value[value.length - 1] === ']'
+}
+
+function isAsciiWhitespace(char: string) {
+  return char === ' ' || char === '\t' || char === '\n' || char === '\r' || char === '\f'
+}
+
+enum AttributePart {
+  Start,
+  Attribute,
+  Value,
+  Modifier,
+  End,
+}
+
+function parseAttributeSelector(value: string) {
```

Comment:
Don't read too much into this function. This was a temporary replacement for the attribute selector parser. I added a proper commit with an implementation + tests + benchmark. Just needed this to make the tests pass.